src/DI/DIContainer.ts

```diff
@@ -19,24 +29,23 @@ export class DIContainerService {
 
   private registerServices() {
     for (const Service of SERVICES_CLASS_REGISTRY) {
-      if (!container.isRegistered(Service)) {
-        container.register<typeof Service>(Service, {
-          useClass: Service as constructor<typeof Service>,
-        });
-      }
+      // TODO test this first if this works fine
+      // if (this.provider.getService(Service)) {
+      //   continue;
+      // }
+
+      this.container.addSingleton(Service);
     }
```

Comment:
in @wroud/di you can use `modules` api to split modules groups registration across application, it's also has it's own regisrty

Reply:
you can get more details here: https://wroud.dev/guide/package/di/scaling/integration#example-core-module